core/domain/html_validation_service.py

```diff
@@ -749,3 +753,51 @@ def _validate_customization_args_in_tag(tag):
                             yield err_msg
     except Exception as e:
         yield str(e)
+
+
+def add_dimensions_to_noninteractive_image_tag(exp_id, html_string):
+    """Adds dimensions to all oppia-noninteractive-image tags.
+
+    Args:
+        html_string: str. HTML string in which the dimensions is to be
+            added.
+        exp_id: str. Exploration id.
+
+    Returns:
+        str. Updated HTML string with the dimensions for all
+            oppia-noninteractive-image tags.
+    """
+    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        filename = unescape_html(image['filepath-with-value'])
+        filename = filename[1:-1]
+        image['filepath-with-value'] = escape_html(json.dumps(
+            get_filename_with_dimensions(filename, exp_id)))
+    return unicode(soup)
+
+
+def get_filename_with_dimensions(filename, exp_id):
+    """Gets the dimensions of the image file.
+
+    Args:
+        filename: str. Name of the file whose dimensions need to be
+            calculated.
+        exp_id: str. Exploration id.
+
+    Returns:
+        dict. A dict representing Fileinfo object of the image.
+    """
+    file_system_class = (
+        fs_domain.ExplorationFileSystem if feconf.DEV_MODE
+        else fs_domain.GcsFileSystem)
+    fs = fs_domain.AbstractFileSystem(file_system_class(exp_id))
+    filepath = (
+        filename if feconf.DEV_MODE
+        else ('image/%s' % filename))
+    content = fs.get(filepath)
+    height, width = gae_image_services.get_image_dimensions(content)
+    filename_wo_filetype = filename[:filename.rfind('.')]
```

Comment:
Find a way to deduplicate this part with regenerate_image_filename_using_dimensions()

Maybe move the latter into here and call it?

Reply:
But we need that function at other places in tests too, and the purpose of the function relates to the name as well as the place it is right now(exp_services). I think it is okay for us to again create the name here.

Reply:
Do not duplicate code.

Reply:
Done